core/src/main/kotlin/com/malinskiy/marathon/analytics/LocalTracker.kt

```diff
@@ -33,8 +29,7 @@ class LocalTracker(private val fileManager: FileManager,
     }
 
     override fun trackDeviceConnected(poolId: DevicePoolId, device: Device) {
-        val json = gson.toJson(device.toDeviceInfo())
-        fileManager.createFile(FileType.DEVICE_INFO, poolId, device).writeText(json)
+        deviceInfoSerializer.deviceConnected(poolId, device)
```

Comment:
@tagantroy I think that we need an aggregating Tracker, so that everything else can subscribe to specific events. Something like a list of subtrackers (e.g. Local + Remote), DelegatingTracker, Adapter class with empty implementation for all track methods. Then each new tracker extends adapter, we register it in the DelegatingTracker and we can process the results separately.